index.js

```diff
@@ -25,19 +25,23 @@ module.exports = {
     else {
 
 
-      const data = await fetch(instagramGraphUrl)
+      const data = await fetch(instagramGraphUrl, {
+        headers: {
+          cookie: `sessionid=${inputs.sessionID}`
+        }
+      })
         .then(res => {
           // ensure that we are only acting on JSON responses
           if(res.headers.get('content-type').includes('application/json')){
             return res.json();
           } else {
-            return null;
+            return res.text();
           }
         });
 
       // If we didn't receive JSON, fail the plugin but not the build
-      if(!data){
-        utils.build.failPlugin(`The Instagram feed did not return JSON data.\nProceeding with the build without the data from the plugin.`);
+      if(!data?.graphql?.user?.edge_owner_to_timeline_media?.edges){
```

Comment:
The change above also leads to this change; `data` can be a text value and it will now fail this check, which is more specific to what the block actually _needs_.